app/models/tag.rb

```diff
@@ -0,0 +1,8 @@
+class Tag < ApplicationRecord
+  has_many :ebook_tags
+  has_many :ebooks, through: :ebook_tags
```

Comment:
Só um comentário: neste momento as tags apenas funcionam com ebooks. Como é que poderiamos alterar isto para que as tags fossem completamente independentes do modelo a que estão associadas? Por exemplo, como é que poderiamos alterar isto para fazer com que users também posso ter tags? (sem criar novas tabelas na base de dados)

Reply:
Através de polimorfismo? Mas teria de manter a join table para poder manter a associação many-to-many.